### PR TITLE
feat: wallet

### PR DESCRIPTION
**Pull request by shreyasananth-3**

### CODE REVIEW

File: packages/cysync-core/src/bgTask/dbListener/helper.ts [hunk 1]
```diff
@@ -27,11 +27,17 @@ const createFuncWithErrorHandler =
   };
 
 const syncWalletsDb = createFuncWithErrorHandler('syncWalletsDb', async () => {
-  const db = getDB();
-
-  const wallets = await db.wallet.getAll(undefined, {
-    sortBy: { key: 'name' },
-  });
+  // const db = getDB();
+
```
Comment (shreyasananth-3) [inline packages/cysync-core/src/bgTask/dbListener/helper.ts:31]: In final PR will replace with original content

File: packages/ui/src/components/molecules/Wallet/WalletHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+}) => (
+  <WalletHeaderWrapper>
+    <Breadcrumb
+      currentPage={title}
+      breadcrumb={breadcrumb}
+      dropdown={dropdown}
+    />
+    <Flex gap={24}>
+      <Button variant="primary">{btnAddToken}</Button>
+      <Button variant="primary">{btnAddAccount}</Button>
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/Wallet/WalletHeader.tsx]: Missing onClick handlers

File: packages/ui/src/components/molecules/Topbar.tsx [hunk 3]
```diff
@@ -95,9 +106,14 @@ export const Topbar: FC<ITopbar> = ({
...
-      <Typography variant="h4" $fontWeight="semibold" color="silver">
-        <LangDisplay text={title} />
-      </Typography>
+      <TitleStyle>
+        <Typography variant="h4" $fontWeight="semibold" color="silver">
+          <LangDisplay text={title} />
+        </Typography>
+        <IconStyle>
+          <PushpinBold />
+        </IconStyle>
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/Topbar.tsx]: 1. Should be a button
2. Should be only present on Wallet page, hence should be made optional
3. Add onClick events via props

File: packages/ui/src/components/molecules/SkeletonLoader.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      <Typography $fontSize={24} $fontWeight="medium">
+        {text}
+      </Typography>
+      <Typography $fontSize={16} color="muted">
+        {subText}
+      </Typography>
+    </Flex>
+    <Flex gap={24}>
+      <Button variant="primary">{$buttonOne}</Button>
+      {$buttonTwo && <Button variant="primary">{$buttonTwo}</Button>}
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/SkeletonLoader.tsx]: onClick handlers?

File: packages/ui/src/components/molecules/Wallet/WalletHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  btnAddAccount?: string;
+  dropdown: DropdownItem[];
+}
+
+const WalletHeaderWrapper = styled.div`
+  display: flex;
+  justify-content: space-between;
+  padding: 10px 20px;
+  margin-top: 16px;
+`;
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/Wallet/WalletHeader.tsx]: No need for styled.div, `Flex` component can be used instead

File: packages/ui/src/components/molecules/Table/WalletTable/index.ts [hunk 6]
```diff
@@ -0,0 +1,2 @@
+export * from './TableRow';
+export * from './TableHeaderWallet';
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/Table/WalletTable/index.ts]: Rename these to `Account Table`

File: packages/ui/src/components/molecules/Table/WalletTable/TableRow.tsx [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+  $rowIndex,
+  $hide,
+  $show,
+  $balance,
+}) => {
+  const { isOpen, toggleAccordion } = useAccordion();
+  return (
+    <RowStyle>
+      <RowContainer $rowIndex={$rowIndex}>
+        <AccountContainer
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/Table/WalletTable/TableRow.tsx]: This needs a row onClick event

File: packages/ui/src/components/molecules/Dropdown/Dropdown.tsx [hunk 8]
```diff
@@ -135,7 +135,12 @@ export const Dropdown: React.FC<DropdownProps> = ({
           }
           onClick={toggleDropdown}
           $restrictedItem
+          leftImage={selectedDropdownItem.leftImage}
+          rightText={selectedDropdownItem.rightText}
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/Dropdown/Dropdown.tsx]: Not needed

File: packages/ui/src/components/molecules/Dropdown/Dropdown.tsx [hunk 9]
```diff
@@ -135,7 +135,12 @@ export const Dropdown: React.FC<DropdownProps> = ({
...
           $hasRightText={!!selectedDropdownItem.rightText}
+          tag={selectedDropdownItem.tag}
+          shortForm={selectedDropdownItem.shortForm}
+          $parentId={selectedDropdownItem.$parentId}
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/Dropdown/Dropdown.tsx]: Not needed

File: packages/ui/src/components/molecules/Breadcrumb.tsx [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+
+export const Breadcrumb: FC<BreadcrumbProps> = ({
+  currentPage,
+  breadcrumb,
+  dropdown,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [focusedIndex, setFocusedIndex] = useState(-1);
+  const dropdownState = () => setIsOpen(!isOpen);
+  const [selectedItem, setSelectedItem] = useState('');
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/Breadcrumb.tsx]: The dropdown state should be in props, just like any other input element

File: packages/ui/src/components/molecules/Dropdown/ListItemDropdown.tsx [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+    }
+  };
+
+  return (
+    <Items
+      onClick={handleBoxClick}
+      checked={checked}
+      $isFocused={focused}
+      tabIndex={0}
+      id={`item-${itemText}`}
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/Dropdown/ListItemDropdown.tsx]: Using the text as `id` is not a good idea. Use a unique identifier instead

File: packages/ui/src/components/molecules/Accordion.tsx [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+  toggleAccordion,
+  $show,
+  $hide,
+}) => (
+  <StyledAccordion onClick={toggleAccordion}>
+    <Typography color="white" $fontSize={14} $fontWeight="medium">
+      {isOpen ? $hide : `${$show} (${tokensLength})`}
+    </Typography>
+    {isOpen ? <ArrowUp /> : <ArrowDown />}
+  </StyledAccordion>
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/Accordion.tsx]: Accordion should be a button

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/molecules/Accordion.tsx (hunk 12 above)
Comment (shreyasananth-3) [reply packages/ui/src/components/molecules/Accordion.tsx]: done

File: packages/ui/src/components/molecules/Dropdown/ListItemDropdown.tsx (hunk 11 above)
Comment (shreyasananth-3) [reply packages/ui/src/components/molecules/Dropdown/ListItemDropdown.tsx]: done

File: packages/ui/src/components/molecules/Breadcrumb.tsx (hunk 10 above)
Comment (shreyasananth-3) [reply packages/ui/src/components/molecules/Breadcrumb.tsx]: done

File: packages/ui/src/components/molecules/Dropdown/Dropdown.tsx (hunk 9 above)
Comment (shreyasananth-3) [reply packages/ui/src/components/molecules/Dropdown/Dropdown.tsx]: done

File: packages/ui/src/components/molecules/Wallet/WalletHeader.tsx (hunk 5 above)
Comment (shreyasananth-3) [reply packages/ui/src/components/molecules/Wallet/WalletHeader.tsx]: Previously designed directly as flex but padding vallues are 10 and 20 px but Flex takes multiples of 8 only, hence designed like that

File: packages/ui/src/components/molecules/Table/WalletTable/TableRow.tsx (hunk 7 above)
Comment (shreyasananth-3) [reply packages/ui/src/components/molecules/Table/WalletTable/TableRow.tsx]: done

File: packages/ui/src/components/molecules/Wallet/WalletHeader.tsx (hunk 2 above)
Comment (shreyasananth-3) [reply packages/ui/src/components/molecules/Wallet/WalletHeader.tsx]: done

File: packages/ui/src/components/molecules/Topbar.tsx (hunk 3 above)
Comment (shreyasananth-3) [reply packages/ui/src/components/molecules/Topbar.tsx]: done

File: packages/ui/src/components/molecules/Dropdown/Dropdown.tsx (hunk 8 above)
Comment (shreyasananth-3) [reply packages/ui/src/components/molecules/Dropdown/Dropdown.tsx]: done

File: packages/ui/src/components/molecules/Table/WalletTable/index.ts (hunk 6 above)
Comment (shreyasananth-3) [reply packages/ui/src/components/molecules/Table/WalletTable/index.ts]: done

File: packages/ui/src/components/molecules/SkeletonLoader.tsx (hunk 4 above)
Comment (shreyasananth-3) [reply packages/ui/src/components/molecules/SkeletonLoader.tsx]: done

File: packages/ui/src/components/molecules/Wallet/WalletHeader.tsx (hunk 5 above)
Comment (irshadCypherock) [reply packages/ui/src/components/molecules/Wallet/WalletHeader.tsx]: You can pass `pr="10px"` etc

File: packages/ui/src/components/molecules/SkeletonLoader.tsx [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+      </Typography>
+      <Typography $fontSize={16} color="muted">
+        {subText}
+      </Typography>
+    </Flex>
+    <Flex gap={24}>
+      <Button variant="primary" onClick={onClick}>
+        {$buttonOne}
+      </Button>
+      {$buttonTwo && <Button variant="primary">{$buttonTwo}</Button>}
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/SkeletonLoader.tsx]: Missing onClick handler

File: packages/ui/src/components/molecules/Dropdown/ListItemDropdown.tsx [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+}) => {
+  const handleBoxClick = () => {
+    if (checkType && checkType === 'radio') {
+      onChange();
+    }
+  };
+
+  useEffect(() => {
+    if (focused) {
+      const divElement = document.getElementById(`item-${itemText}`);
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/Dropdown/ListItemDropdown.tsx]: I think this needs to be changed to `id` as well

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/molecules/SkeletonLoader.tsx (hunk 13 above)
Comment (shreyasananth-3) [reply packages/ui/src/components/molecules/SkeletonLoader.tsx]: done

File: packages/ui/src/components/molecules/Dropdown/ListItemDropdown.tsx (hunk 14 above)
Comment (shreyasananth-3) [reply packages/ui/src/components/molecules/Dropdown/ListItemDropdown.tsx]: done

File: packages/ui/src/components/molecules/Table/TableSearch.tsx [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+`;
+export const MainAppBody = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  gap: 16px;
+  position: absolute;
+  overflow: auto;
+`;
+
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/Table/TableSearch.tsx]: Why is main app body and wrapper inside `TableSearch` component?

File: packages/ui/src/components/molecules/Topbar.tsx [hunk 16]
```diff
@@ -83,6 +88,12 @@ export const Topbar: FC<ITopbar> = ({
     error: <SyncProblem fill={theme?.palette.warn.main} />,
   };
 
+  const TitleStyle = styled.div`
+    display: flex;
+    align-items: center;
+    gap: 16px;
+  `;
```
Comment (irshadCypherock) [inline packages/ui/src/components/molecules/Topbar.tsx]: Move this out of the component to solve this error 
image

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/molecules/Table/TableSearch.tsx (hunk 15 above)
Comment (shreyasananth-3) [reply packages/ui/src/components/molecules/Table/TableSearch.tsx]: Needs to go in a separate file.

**Pull request closed without merging**